### PR TITLE
v0.15.2 release backports

### PR DESCRIPTION
**Pull request by navidshaikh**
## Description
client v0.15.2 release backports with a bug fix and pinning go 1.14 in go.mod (inline with serving v0.15.x release).

## Changes
* Backport from master:
  -  #866 
  -  #880 
  -  #883
* CHANGELOG update

### CODE REVIEW

**Comment by navidshaikh** [comment]
@danielhelfand @dprotaso : Can you please add consent for CLA check ?

**Comment by navidshaikh** [comment]
@googlebot I consent. 

/retest

**Comment by danielhelfand** [comment]
> @danielhelfand @dprotaso : Can you please add consent for CLA check ?

@googlebot I consent.

File: hack/build.sh [hunk 1]
```diff
@@ -16,6 +16,9 @@
 
 set -o pipefail
 
+[[ ! -v REPO_ROOT_DIR ]] && REPO_ROOT_DIR="$(git rev-parse --show-toplevel)"
+readonly REPO_ROOT_DIR
```
Comment (rhuss) [inline hack/build.sh:20]: That won't work on macOS. You would need to backport my fix, too then.

Comment (navidshaikh) [reply hack/build.sh]: fixed

**Pull request merged**